src/librustc_codegen_ssa/mir/mod.rs

```diff
@@ -234,7 +234,7 @@ pub fn codegen_mir<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>>(
         debug_context,
     };
 
-    let memory_locals = analyze::non_ssa_locals(&fx);
+    let memory_locals = analyze::non_ssa_locals(&fx, &mut mir);
```

Comment:
In case the analysis definitely needs the body cache, you could just pre-compute the body cache and pass in an immutable reference